backend/multi/backend.c

```diff
@@ -57,27 +68,19 @@ struct wlr_backend_impl backend_impl = {
 	.get_egl = multi_backend_get_egl,
 };
 
-static void handle_display_destroy(struct wl_listener *listener, void *data) {
-	struct wlr_multi_backend *backend =
-		wl_container_of(listener, backend, display_destroy);
-	multi_backend_destroy(&backend->backend);
-}
-
-struct wlr_backend *wlr_multi_backend_create(struct wl_display *display,
-		struct wlr_session *session) {
+struct wlr_backend *wlr_multi_backend_create(struct wl_display *display) {
 	struct wlr_multi_backend *backend =
 		calloc(1, sizeof(struct wlr_multi_backend));
 	if (!backend) {
 		wlr_log(L_ERROR, "Backend allocation failed");
 		return NULL;
 	}
 
-	backend->session = session;
 	wl_list_init(&backend->backends);
 	wlr_backend_init(&backend->backend, &backend_impl);
 
-	session->display_destroy.notify = handle_display_destroy;
-	wl_display_add_destroy_listener(display, &session->display_destroy);
+	wl_signal_init(&backend->events.backend_add);
```

Comment:
Why don't you listen to display destroy anymore? We need to free all backends, including the multi-backend, when the display is destroyed.

Reply:
My intention is that the drm backend should be able to added and removed at runtime, and I thought session destroy would emit the destroy event on the drm backend and have it removed at that time.

Reply:
But you still have to free the multi-backend